File: leet/0686-repeated-string-match/repeated-string-match.py
class Solution:
    def repeatedStringMatch(self, A: str, B: str) -> int:
        a = A
        while len(a) <= len(A) * 2 + len(B):
            if a.find(B) == -1:
                a += A 
            else:
                return len(a) // len(A)
        
        return -1

File: leet/0686-repeated-string-match/repeated-string-match.js
/**
 * @param {string} A
 * @param {string} B
 * @return {number}
 */
var repeatedStringMatch = function(A, B) {
    let a = A.length
    let b = B.length

    if (b <= a){
        if (A.indexOf(B) != -1) return 1
        if ((A + A).indexOf(B) != -1) return 2
        return -1
    }

    let x = Math.floor(b / a)
    let str = new Array(x).fill(A).join("")
    if(str.indexOf(B) != -1) return x
    str += A
    if(str.indexOf(B) != -1) return x + 1
    str += A
    if(str.indexOf(B) != -1) return x + 2
    return -1
};
